core/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -66,40 +71,68 @@
             + Long.BYTES // offset
             + Long.BYTES; // generation
 
-    Checkpoint(long offset, int numOps, long generation, long globalCheckpoint) {
+    /**
+     * Create a new translog checkpoint.
+     *
+     * @param offset           the current offset in the translog
+     * @param numOps           the current number of operations in the translog
+     * @param generation       the current translog generation
+     * @param minSeqNo         the current minimum sequence number of all operations in the translog
+     * @param maxSeqNo         the current maximum sequence number of all operations in the translog
+     * @param globalCheckpoint the last-known global checkpoint
+     */
+    Checkpoint(long offset, int numOps, long generation, long minSeqNo, long maxSeqNo, long globalCheckpoint) {
         this.offset = offset;
         this.numOps = numOps;
         this.generation = generation;
+        this.minSeqNo = minSeqNo;
+        this.maxSeqNo = maxSeqNo;
```

Comment:
Or maybe something like `minSeqNo <= maxSeqNo || minSeqNo == Long.MAX_VALUE && maxSeqNo == Long.MIN_VALUE` if we want to keep the merging easier.